### ISSUE TITLE
Declaring dependencies in the root project not working

### ISSUE THREAD

**Issue by d-eder** [cloudflightio/autoconfigure-gradle-plugin#183]
I'm using the autoconfigure-gradle on a multii module project. When the dependencies for the submodule are declared in the root module I get the error: 
``` 
file:///C:/src/demo/build.gradle.kts:16:9: Unresolved reference: implementation
```

To workaround this I'd have to apply the kotlin plugin on the root project manually.

I created a temporary repo to reproduce the issue:  https://github.com/d-eder/tmp-demo

**Comment by cgrabmann** [comment]
This is a limitation by Gradle.

Please read the Kotlin-DSL documentation for [Type-Safe-Accessors](https://docs.gradle.org/current/userguide/kotlin_dsl.html#type-safe-accessors) and [Multi-Project-Builds](https://docs.gradle.org/current/userguide/kotlin_dsl.html#sec:multi_project_builds)

TLDR: The Gradle Kotlin Script gets compiled to be runnable but since the "Root" script cannot know which plugins get applied to the Sub-Projects it does not generate the type-safe-accessors. That's why you cannot use implementation as a method here.

There are 2 ways to fix this:
1. You can use a `String` as "method": `"implementation"(platform("org.springframework.boot:spring-boot-dependencies:3.1.4"))`
2. You can use the Groovy-Version of Gradle-Scripts, since there the `implementation` method is dynamically resolved (like it is when you use a `String` as "method" in the Kotlin-Script)
